packages/credentials-interface/src/eip712proof.ts

```diff
@@ -0,0 +1,18 @@
+import { TypedDataField } from '@ethersproject/abstract-signer';
+
+export interface EIP712Proof {
+  // ! This is a hack to get the type to work with PEX types. PEX is not supporting EIP712 signatures.
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  [x: string]: string | string[] | any;
+  '@context': string | string[];
+  type: string;
+  proofPurpose: string;
+  proofValue: string;
+  verificationMethod: string;
+  created: string;
```

Comment:
What about this?
```suggestion
export interface EIP712Proof extends IProof {
  // ! Need to add index signature to be compatible with PEX type
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  [x: string]: string | string[] | any;
```

Reply:
@Harasz Doesn't look like this is implemented yet. Did you try it and it doesn't work?

Reply:
Extending to IProof? I think it doesn’t make sense. This is interface just for eip712 proof, bc PEX doesn’t has one. Presentation and credential proof object has type Iproof | EIP712.

Reply:
Hmm, well my rational for extending `IProof` is that I see EIP712Proof as being IProof but with additional properties so from a DRY standpoint, it doesn't make sense to duplicate the IProof props in EIP712Proof. Is it is IProof has props that aren't possible to have in EIP712Proof?

Even if extending, I would keep `credentialProof: IProof | EIP712` because to me that is saying that credentialProof can be either `IProof` or `EIP712Proof`, which is correct

Reply:
It will suggest that `EIP712Proof` can have a `jws` property because `IProof` has one. I think it can be misleading, don't you think?

Reply:
Ok, good point. Let's keep as is then 👍